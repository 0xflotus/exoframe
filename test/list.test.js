/* eslint-env jest */
// mock config for testing
jest.mock('../src/config', () => require('./__mocks__/config'));

// npm packages
const nock = require('nock');
const sinon = require('sinon');

// our packages
const {handler: list} = require('../src/commands/list');

const services = [
  {
    Id: '123',
    Name: '/test',
    Config: {
      Labels: {'traefik.frontend.rule': 'Host:test.host', 'exoframe.project': 'test'},
    },
    State: {
      Status: 'Up 10 minutes',
    },
    NetworkSettings: {
      Networks: {
        exoframe: {
          Aliases: null,
        },
      },
    },
  },
  {
    Id: '321',
    Name: '/test2',
    Config: {
      Labels: {'exoframe.project': 'test'},
    },
    State: {
      Status: 'Up 12 minutes',
    },
    NetworkSettings: {
      Networks: {
        exoframe: {
          Aliases: null,
        },
      },
    },
  },
  {
    Id: '111',
    Name: '/test3',
    Config: {
      Labels: {'exoframe.project': 'other'},
    },
    State: {
      Status: 'Up 13 minutes',
    },
    NetworkSettings: {
      Networks: {
        exoframe: {
          Aliases: null,
        },
      },
    },
  },
];

// test removal
test('Should get list of deployments', done => {
  // handle correct request
  const listServer = nock('http://localhost:8080')
    .get(`/list`)
    .reply(200, services);
  // spy on console
  const consoleSpy = sinon.spy(console, 'log');
  // execute login
  list().then(() => {
    // make sure log in was successful
    // check that server was called
    expect(listServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // restore console
    console.log.restore();
    listServer.done();
    done();
  });
});
