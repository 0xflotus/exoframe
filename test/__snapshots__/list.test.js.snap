// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should get list of deployments 1`] = `
Array [
  Array [
    "[32m3 deployments found on http://localhost:8080:[39m
[32m[39m",
  ],
  Array [
    "Deployments for [1mtest[22m:",
  ],
  Array [],
  Array [
    "[31m   ID      [39m[90m [39m[31m   URL         [39m[90m [39m[31m   Hostname   [39m[90m [39m[31m   Status          [39m
   test    [90m [39m   test.host   [90m [39m   Not set    [90m [39m   Up 10 minutes   
   test2   [90m [39m   Not set     [90m [39m   Not set    [90m [39m   Up 12 minutes   ",
  ],
  Array [],
  Array [
    "Other deployments:",
  ],
  Array [],
  Array [
    "[31m   ID      [39m[90m [39m[31m   URL       [39m[90m [39m[31m   Hostname   [39m[90m [39m[31m   Status          [39m
   test3   [90m [39m   Not set   [90m [39m   Not set    [90m [39m   Up 13 minutes   ",
  ],
]
`;
